lib/libutils/ext/trace.c

```diff
@@ -68,59 +67,69 @@ static const char *trace_level_to_string(int level, bool level_ok)
 }
 
 /* Format trace of user ta. Inline with kernel ta */
-void trace_printf(const char *function, int line, int level, bool level_ok,
-		  const char *fmt, ...)
+void trace_printf(const char __maybe_unused *function, int __maybe_unused line,
+		  int level, bool level_ok, const char *fmt, ...)
 {
 	va_list ap;
 	char buf[MAX_PRINT_SIZE];
 	size_t boffs = 0;
 	int res;
-	int thread_id;
+	int __maybe_unused thread_id;
 
 	if (level_ok && level > trace_level)
 		return;
 
-	res = snprintk(buf, sizeof(buf), "%s ",
+	/* Print the type of message */
+	res = snprintk(buf, sizeof(buf), "%s/",
 		       trace_level_to_string(level, level_ok));
 	if (res < 0)
 		return;
 	boffs += res;
 
+	/* Print the location, i.e., TEE core or TA */
+	res = snprintk(buf + boffs, sizeof(buf) - boffs, "%s:",
+		       trace_ext_prefix);
+	if (res < 0)
+		return;
+	boffs += res;
+
+	/* Print the Thread ID */
 	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
 		thread_id = -1;
 	else
 		thread_id = trace_ext_get_thread_id();
 
 	if (thread_id >= 0) {
-		res = snprintk(buf + boffs, sizeof(buf) - boffs, "[0x%x] ",
+#if CFG_NUM_THREADS > 9
+		res = snprintk(buf + boffs, sizeof(buf) - boffs, "%02d ",
 			       thread_id);
+#else
+		res = snprintk(buf + boffs, sizeof(buf) - boffs, "%01d ",
+			       thread_id);
+#endif
+		if (res < 0)
+			return;
+		boffs += res;
+	} else {
+		/* If there is no thread ID, then fill in with blanks */
+		res = snprintk(buf + boffs, sizeof(buf) - boffs, "   ");
```

Comment:
The thread id printing could be simplified (and correct with regards to the number of blanks) with a helper function like this:
```
static int print_thread_id(char *buf, size_t bs, int thread_id)
{
#if CFG_NUM_THREADS > 9
        int num_thread_digits = 2;
#else
        int num_thread_digits = 1;
#endif

        if (thread_id >= 0)
                return snprintk(buf, bs, "%0*d", num_thread_digits, thread_id);
        else
                return snprintk(buf, bs, "%*s", num_thread_digits, "");
}
```